openmdao/drivers/genetic_algorithm_driver.py

```diff
@@ -511,6 +511,8 @@ def objective_callback(self, x, icase):
                     elif (con['equals'] is not None) and np.any(np.abs(con['equals']) < almost_inf):
                         diff = val - con['equals']
                         violation = np.absolute(diff)
+                    else:
+                        violation = np.zeros(val.shape)
```

Comment:
Since the violation in this case is all zeros, it seems unnecessary that we would create a new array of some potentially large size since it won't contribute anything to the 'fun' value.   I see a number of other performance issues in other parts of this code not related to this PR, so it might be worthwhile to add a separate issue to correct them.